frontend/server/controllers/NotificationController.php

```diff
@@ -23,5 +23,31 @@ public static function apiMyList(Request $r) {
         ];
     }
 
-    //TODO: apiReadNotifications
+    /**
+     * Updates notifications as read in database
+     *
+     * @param Request $r
+     * @return array
+     * @throws InvalidDatabaseOperationException
+     */
+    public static function apiReadNotifications(Request $r) {
+        self::authenticateRequest($r);
```

Comment:
como necesitas que `$r->user` sea válido, puedes hacer lo mismo que hacemos en `UserController`? https://github.com/omegaup/omegaup/blob/ed5d6d429c4923736164b814e1e8ff4b5a4ee08d/frontend/server/controllers/ContestController.php#L231


```suggestion
        self::authenticateRequest($r, true /* requireMainUserIdentity */);
```